### PR TITLE
add API to save il2cpp data dir for il2cpp on mono

### PR DESCRIPTION
**Pull request by mderoy**

### CODE REVIEW

**Comment by joncham** [comment]
I am not sure about name of `mono_unity_set_il2cpp_data_dir` (use of il2cpp) but in general LGTM.

**Comment by joshpeterson** [comment]
I agree, just remove the `il2cpp_` part of the names.

**Comment by mderoy** [comment]
@joshpeterson I added it since its the "il2cpp data dir" the name seems to make sense to me. Also the naming implies that this function is only needed for il2cpp on mono

**Comment by joshpeterson** [comment]
I would like to avoid the use of references to the code generation in the VM code if we can. There are a number of other functions in unity-utils.c that are specific to IL2CPP on Mono already, so there is a precedent for having specific function there which do not have "il2cpp" in the name.

File: mono/metadata/unity-utils.c [hunk 1]
```diff
@@ -1021,3 +1021,20 @@ mono_unity_install_memory_callbacks (MonoAllocatorVTable* callbacks)
 {
 	mono_set_allocator_vtable (callbacks);
 }
+
+static char* il2cpp_data_dir = NULL;
```
Comment (joshpeterson) [inline mono/metadata/unity-utils.c]: Can you remove the `il2cpp_` from the name of this variable as well?

**Pull request merged**